nomos-services/network/src/backends/libp2p.rs

```diff
@@ -112,14 +115,20 @@ impl NetworkBackend for Libp2p {
                                 log_error!(swarm.connect(peer_id, peer_addr));
                             }
                             Command::Broadcast { topic, message } => {
-                                match swarm.broadcast(&topic, message) {
+                                match swarm.broadcast(&topic, message.clone()) {
                                     Ok(id) => {
                                         tracing::debug!("broadcasted message with id: {id} tp topic: {topic}");
                                     }
                                     Err(e) => {
                                         tracing::error!("failed to broadcast message to topic: {topic} {e:?}");
                                     }
                                 }
+                                log_error!(events_tx.send(Event::Message(Message {
+                                    source: None,
+                                    data: message.into(),
+                                    sequence_number: None,
+                                    topic: TopicHash::from_raw(topic)
+                                })));
```

Comment:
If this is for publishers to receive messages published themselves, I think that we need to check whether the publisher is also subscribed to the topic. In our case, all Nomos nodes are subscribed to a common topic. But, from the perspective of libp2p-backend, the publisher might be able to broadcast messages without being subscribed to the topic, IIRC: https://docs.libp2p.io/concepts/pubsub/overview/#fan-out.

It seems that we can list up all topics that we are currently subscribed to, as below:
```rust
self.swarm.behaviour_mut().gossipsub.topics()
```

Reply:
I just found the libp2p gossipsub [`allow_self_origin`](https://github.com/libp2p/rust-libp2p/blob/3c5940aeadb9ed8527b6f7aa158797359085293d/protocols/gossipsub/src/config.rs#L261-L261) option. By enabling this, it seems that we can receive messages published by us. But, according to the function comment in the link above and [their implementation](https://github.com/libp2p/rust-libp2p/blob/3c5940aeadb9ed8527b6f7aa158797359085293d/protocols/gossipsub/src/behaviour.rs#L1722C11-L1722C11), enabling this option may prevent penalizing peers who send our own messages back. 

So, I think it would be safer to just check if we're subscribed to the topic or not.

Reply:
I just opened a PR for this: https://github.com/logos-co/nomos-node/pull/292

Reply:
thanks!